project-api/test-artifacts/sample-openapi-build-tool/src/main/java/build/tool/runner/diagnostics/ToolDiagnostic.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package build.tool.runner.diagnostics;
+
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticProperty;
+import io.ballerina.tools.diagnostics.Location;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextRange;
+
+import java.util.List;
+
+/**
+ * A sample diagnostic class for a sample build tool.
+ *
+ * @since 2201.9.0
+ */
+public class ToolDiagnostic extends Diagnostic {
```

Comment:
We don't need to introduce a new diagnostic type for this. `io.ballerina.tools.diagnostics.DiagnosticFactory.createDiagnostic` is an API that we have already provided to create diagnostics. Tool authors can use this to create the diagnostic.